src/includes/class-alma-wc-payment-gateway.php

```diff
@@ -231,25 +237,82 @@ public function get_icon() {
 	 */
 	public function process_admin_options() {
 		$previously_saved = parent::process_admin_options();
+		set_transient( 'alma-admin-soc-panel', true, 5 );
+		$post_data = $this->get_post_data();
+
+		$this->process_checkout_legal( $post_data );
+
+		// Check if the live_api_key has changed.
+		if (
+			alma_wc_plugin()->settings->live_api_key !== $post_data['woocommerce_alma_live_api_key']
+		) {
+			$this->settings['share_of_checkout_enabled_date'] = '';
+		}
+
+		// Check if the mode has changed to live (by default in the module you are in test so the legal is not init).
+		if (
+			'live' === $post_data['woocommerce_alma_environment']
+			&& 'test' === alma_wc_plugin()->settings->get_environment()
+		) {
+			$legal = new Alma_WC_Admin_Helper_Check_Legal();
+			$legal->init();
+		}
+
+		if (
+			'test' === $post_data['woocommerce_alma_environment']
+			&& 'live' === alma_wc_plugin()->settings->get_environment()
+		) {
+			delete_transient( 'alma-admin-soc-panel' );
+		}
 
 		$this->convert_amounts_to_cents();
 		$this->update_settings_from_merchant();
 		alma_wc_plugin()->settings->update_from( $this->settings );
 		alma_wc_plugin()->force_check_settings();
 
-		$value = $this->settings;
-		/** LEGAL CHECKOUT FEATURE */
-		$post_data = $this->get_post_data();
+		$values = $this->settings;
```

Comment:
check if inlining `$this->settings` is possible (`$values` not seems useful here)

Reply:
done